src/scripts/main.js

```diff
@@ -1,11 +1,26 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    const handleEvent = (e) => {
+      element.removeEventListener(eventName, handleEvent);
+
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
```

Comment:
It's a good practice to avoid using template literals when there's no dynamic content inside. In this case, you can simply use a string. Also, it's better to return the event object itself from the Promise, so the caller function can use any property of the event object, not only the event name, element name and id. This will make your function more flexible and usable.